src/main/java/com/google/sps/data/Journey.java

```diff
@@ -79,45 +87,184 @@ public int getNumberOfWaypoints() {
   /**
    * Returns true if the minimum time journey fits in the timeslot and false otherwise.
    */
-  public boolean findJourneyForTimeslot(DeliverySlot deliverySlot) throws ApiException, IOException, InterruptedException {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+  public boolean findJourneyForTimeslot(DeliverySlot deliverySlot) throws ApiException, DataNotFoundException, IOException, InterruptedException {
+    ArrayList<ArrayList<Pair>> bestTime = findOptimalTimePaths();
+    int numberOfWaypoints = getNumberOfWaypoints();
+    int allWaypointsConfig = (1 << numberOfWaypoints) - 1;
+    int minTime = bestTime.get((1 << numberOfWaypoints)).get(0).first;
+    for (int waypointIndex = 0; waypointIndex < numberOfWaypoints; ++waypointIndex) {
+      if (bestTime.get(allWaypointsConfig).get(waypointIndex).first < minTime) {
+        minTime = bestTime.get(allWaypointsConfig).get(waypointIndex).first;
+      }
+    }
+
+    long deliverySlotDurationInSeconds = Duration.between(deliverySlot.getStartTime().toInstant(),
+        deliverySlot.getEndTime().toInstant()).toMillis() / 1000;
+
+    return (minTime <= (int)deliverySlotDurationInSeconds);
   }
 
   /**
    * Computes and returns the optimal order in which points should be visited starting from the
    * last point to the fist one (the start Point).
```

Comment:
I'm a bit confused why do you return from last to first.

Why not just return the right order (e.g. orderedWaypoints.reverse())?

Reply:
I changed it to the right order.